### PR TITLE
Fix toggleable block opening sounds (doors, trapdoors, fence gates)

### PR DESCRIPTION
**Pull request by letsgoawaydev**
Fixes the trapdoor, door and fence gate sounds to be the material specific sounds.

~~Requires https://github.com/GeyserMC/mappings/pull/96 to be accepted as the copper door sounds are fixed there.~~ This has been merged.

~~Note that the cherry fence gate sometimes does not play the sound, this is a bug in Bedrock Edition's `sounds_definitions.json`, see https://github.com/Mojang/bedrock-samples/pull/60~~ https://github.com/GeyserMC/GeyserOptionalPack/pull/56 should resolve this

Copper sounds have not been done as it seems like copper waxing and scrape events do not work in geyser at the moment.

Tested with Geyser-Spigot on 1.21 PaperMC

### CODE REVIEW

**Comment by letsgoawaydev** [comment]
Accidentally forgot to sync so it broke everything, reopening

File: core/src/main/java/org/geysermc/geyser/translator/sound/block/FenceGateSoundInteractionTranslator.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import org.geysermc.geyser.translator.sound.BlockSoundInteractionTranslator;
+import org.geysermc.geyser.translator.sound.SoundTranslator;
+import org.geysermc.geyser.util.SoundUtils;
+
+@SoundTranslator(blocks = {"fence_gate"})
+public class FenceGateSoundInteractionTranslator implements BlockSoundInteractionTranslator {
+    @Override
+    public void translate(GeyserSession session, Vector3f position, String identifier) {
+        if (identifier.contains("iron")) return;
+        GeyserImpl.getInstance().getLogger().info(identifier);
```
Comment (onebeastchris) [inline core/src/main/java/org/geysermc/geyser/translator/sound/block/FenceGateSoundInteractionTranslator.java:40]:
```suggestion
```

File: core/src/main/java/org/geysermc/geyser/translator/sound/block/FenceGateSoundInteractionTranslator.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import org.geysermc.geyser.session.GeyserSession;
+import org.geysermc.geyser.translator.sound.BlockSoundInteractionTranslator;
+import org.geysermc.geyser.translator.sound.SoundTranslator;
+import org.geysermc.geyser.util.SoundUtils;
+
+@SoundTranslator(blocks = {"fence_gate"})
+public class FenceGateSoundInteractionTranslator implements BlockSoundInteractionTranslator {
+    @Override
+    public void translate(GeyserSession session, Vector3f position, String identifier) {
+        if (identifier.contains("iron")) return;
```
Comment (onebeastchris) [inline core/src/main/java/org/geysermc/geyser/translator/sound/block/FenceGateSoundInteractionTranslator.java:39]: Is this check needed here?

**Review by onebeastchris — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/geysermc/geyser/translator/sound/block/FenceGateSoundInteractionTranslator.java (hunk 2 above)
Comment (letsgoawaydev) [reply core/src/main/java/org/geysermc/geyser/translator/sound/block/FenceGateSoundInteractionTranslator.java:39]: No, this was a mistake, I forgot to delete this before opening a pull request, sorry. I'll make a commit now

Comment (letsgoawaydev) [reply core/src/main/java/org/geysermc/geyser/translator/sound/block/FenceGateSoundInteractionTranslator.java]: Its now removed

**Review by onebeastchris — APPROVED** [review]
Looks good to me, thanks! 
I wonder if we can fix the cherry fence gate sound issue by changing the sounds used in the optionalpack

**Comment by letsgoawaydev** [comment]
> Looks good to me, thanks! I wonder if we can fix the cherry fence gate sound issue by changing the sounds used in the optionalpack

https://github.com/GeyserMC/GeyserOptionalPack/pull/56 should work

**Comment by letsgoawaydev** [comment]
Just realised the waxing sound for the copper door also needs to be fixed. Will do that now

**Comment by letsgoawaydev** [comment]
Never mind, it seems like the copper effects even for other copper blocks aren't working so if that gets changed it should be done in another pr

**Comment by letsgoawaydev** [comment]
This is marked as draft by Chris as he has found that there are issues with the event not triggering the door to open properly. We need to send the blocks runtime ID in the extra data section of the Level Event Packet for the sounds to work properly instead of just playing the sound otherwise the door doesnt open correctly.

**Comment by letsgoawaydev** [comment]
Marking as ready for review as the code has been changed to not rely on the Sound ID's and is now correctly using the SoundEvents and extra data

File: core/src/main/java/org/geysermc/geyser/translator/sound/block/OpenableSoundInteractionTranslator.java [hunk 3]
```diff
@@ -26,22 +26,40 @@
...
+    private SoundEvent getSound(boolean open, String identifier) {
+        if (identifier.contains("_door")) {
+            return open ? SoundEvent.DOOR_OPEN : SoundEvent.DOOR_CLOSE;
+        }
+        else if (identifier.contains("_trapdoor")){
+            return open ? SoundEvent.TRAPDOOR_OPEN : SoundEvent.TRAPDOOR_CLOSE;
+        }
+        else { // Fence Gate
+            return open ? SoundEvent.FENCE_GATE_OPEN : SoundEvent.FENCE_GATE_CLOSE;
+        }
```
Comment (onebeastchris) [inline core/src/main/java/org/geysermc/geyser/translator/sound/block/OpenableSoundInteractionTranslator.java:63]:
```suggestion
        if (identifier.contains("_door")) {
            return open ? SoundEvent.DOOR_OPEN : SoundEvent.DOOR_CLOSE;
        }
        
        if (identifier.contains("_trapdoor")) {
            return open ? SoundEvent.TRAPDOOR_OPEN : SoundEvent.TRAPDOOR_CLOSE;
        }
        
        // Fence Gate
        return open ? SoundEvent.FENCE_GATE_OPEN : SoundEvent.FENCE_GATE_CLOSE;
```
Small nitpick, as per https://github.com/GeyserMC/Geyser/blob/master/CONTRIBUTING.md

File: core/src/main/resources/mappings
Comment (onebeastchris) [inline core/src/main/resources/mappings:1]: Can you update the mappings once more to include https://github.com/GeyserMC/mappings/pull/95?

**Review by onebeastchris — COMMENTED** [review]
Looks good to me! I'll do some testing & add a final review later to double-check these; but nice work!

File: core/src/main/resources/mappings
Comment (letsgoawaydev) [reply core/src/main/resources/mappings:1]: Sure!

**Pull request merged**